add_nyt_bestseller_tag.py

```diff
@@ -0,0 +1,111 @@
+"""
+A bot to add a NYT bestseller tag 'nyt:{list_name_encoded}={published_date}' to each book from the list of isbns.
+The list of bestsellers to add is a json file of the following format:
+[
+    {
+        "list_name_encoded": "business-books",
+        "published_date": "2020-07-12",
+        "isbns": [
+            "9780735211292",
+            "9780399592522"
+        ]
+    },
+]
+This script would be called from the command line like so:
+$ python add_nyt_bestseller_tag.py --file=~/nyt_bstslrs.json --limit=1 --dry-run=True
+NOTE: This script checks if there is a tag on a book that starts with
+'nyt:' and if the book has such a tag the script does nothing.
+If there is no such book with requested isbn in the OL, the script makes request
+by the book isbn https://openlibrary.org/isbn/{isbn} , triggering autoimport
+"""
+
+import json
+import requests
+
+from olclient.bots import AbstractBotJob
+
+
+class AddNytBestsellerJob(AbstractBotJob):
+    @staticmethod
+    def need_to_add_nyt_bestseller_tag(work)-> bool:
+        """Returns False if the book already has
+        a tag that starts with 'nyt:'"""
+        for subj in work['subjects']:
+            if subj.startswith(('nyt:', 'nytimes:')):
+                return False
+        return True
+
+    @staticmethod
+    def add_tag(work, new_tag) -> None:
+        """Adds a new tag to a work"""
+        if 'subjects' not in work:
+            work['subjects'] = list(new_tag)
+        else:
+            work['subjects'].append(new_tag)
+
+    def book_exists_in_ol(self, book_isbn) -> bool:
+        """Returns True if Edition with book_isbn exists.
+        Returns False otherwise"""
+        try:
+            found_edition = self.ol.Edition.get(isbn=book_isbn)
+            if not found_edition:
+                self.logger.info('Not found edition for ISBN' + book_isbn)
+                return False
+            self.logger.info('Found edition for ISBN' + book_isbn)
+            return True
+        except Exception as e:
+            # TODO check the exception type
+            job.logger.exception('There is no book with ISBN ' + book_isbn)
+            return False
+
+    def request_book_import_by_isbn(self, book_isbn) -> None:
+        """ Makes request to the book_isbn
+        https://openlibrary.org/isbn/{book_isbn} """
+        url = "https://openlibrary.org/isbn/%s" % (book_isbn)
+        try:
+            requests.get(url)
```

Comment:
Just to confirm the request is made even in the dry run, not sure if this side effect is desired.

Reply:
it's fine for the get request to be made during dry run. imo dry run is just about not making edits just seeing what edits should be made

Reply:
Got it, thanks! My concern was that even dry run changes state of the system (OL). I could check if the run is dry and not to make the request, but let's leave it as is for now.